salt/states/win_dacl.py

```diff
@@ -0,0 +1,182 @@
+        - copy_inherited_acl: False
+'''
+
+
+__virtualname__ = 'win_dacl'
+
+
+def __virtual__():
+    '''
+    Load this state if the win_acl module exists
+    '''
+    return 'win_dacl' if 'win_dacl.add_ace' in __salt__ else False
+
+
+def present(name, objectType, user, permission, acetype, propagation):
+    '''
+    Ensure an ACE is present
+    '''
+    ret = {'name': name,
+           'result': True,
+           'changes': {},
+           'comment': []}
+    tRet = __salt__['win_dacl.check_ace'](name, objectType, user, permission, acetype, propagation, True)
+    if tRet['result']:
+        if not tRet['Exists']:
+            addRet = __salt__['win_dacl.add_ace'](name, objectType, user, permission, acetype, propagation)
+            if addRet['result']:
+                ret['result'] = True
+                ret['changes'] = dict(ret['changes'], **addRet['changes'])
+            else:
+                ret['result'] = False
+                ret['comment'] = ret['comment'] + addRet['comment']
+    else:
+        ret['result'] = False
+        ret['comment'] = tRet['comment']
+        return ret
+    return ret
+
+
+def absent(name, objectType, user, permission, acetype, propagation):
+    '''
+    Ensure a Linux ACL does not exist
+    '''
+    ret = {'name': name,
+           'result': True,
+           'changes': {},
+           'comment': []}
+    tRet = __salt__['win_dacl.check_ace'](name, objectType, user, permission, acetype, propagation, True)
+    if tRet['result']:
+        if tRet['Exists']:
+            addRet = __salt__['win_dacl.rm_ace'](name, objectType, user, permission, acetype, propagation)
+            if addRet['result']:
+                ret['result'] = True
+                ret['changes'] = dict(ret['changes'], **addRet['changes'])
+            else:
+                ret['result'] = False
+                ret['comment'] = ret['comment'] + addRet['comment']
+    else:
+        ret['result'] = False
+        ret['comment'] = tRet['comment']
+        return ret
+    return ret
+
+
+def enableinheritance(name, objectType, clear_existing_acl=False):
+    '''
+    Ensure an object in inheriting ACLs from its parent
+    '''
+    ret = {'name': name,
+           'result': True,
+           'changes': {},
+           'comment': []}
+    tRet = __salt__['win_dacl.check_inheritance'](name, objectType)
+    if tRet['result']:
+        if not tRet['Inheritance']:
+            eRet = __salt__['win_dacl.enable_inheritance'](name, objectType, clear_existing_acl)
+            if eRet['result']:
+                ret['result'] = True
+                ret['changes'] = dict(ret['changes'], **eRet['changes'])
+            else:
+                ret['result'] = False
+                ret['comment'] = ret['comment'] + eRet['comment']
+    else:
+        ret['result'] = False
+        ret['comment'] = tRet['comment']
+        return ret
+    return ret
+
+
+def disableinheritance(name, objectType, copy_inherited_acl=True):
```

Comment:
Would this name be better with an underscore? (same with others)
Or could it be shorter? Just enable/disable?